### PR TITLE
Added Director Nav Links

### LINKED ISSUE #50: Add Director Nav Links
image

ignore "DIRECTOR" section

on website, currently just has interview. add additional links to the director navbar.

refer to routes in directors page
refer to /data/directors/navigation.tsx

### PR DESCRIPTION
**Pull request by Jacob-GHub**
Added new routes and ignored the initial "director" section. I followed the routes on figma, but I was unsure if this was the correct route for the "directors" section

Screenshot 2024-11-05 at 2 38 15 PM

The Routes: src > data > director
``` tsx
export const links = [
  {
    name: "Director",
    link: "",
  },
  {
    name: "Applications",
    link: "/director/:program/application",
  },
  {
    name: "Leads",
    link: "/director/leads",
  },
  {
    name: "Directors",
    link: "/director/:program/profile/:uid",
  },
  {
    name: "Assignments",
    link: "/director/:program/assignments",
  },
  {
    name: "View Teams",
    link: "/director/:program/view",
  },
  {
    name: "Email",
    link: "/director/contacts",
  },
];
```

### CODE REVIEW

**Comment by shahdivyank** [comment]
attach issue

File: src/app/director/[program]/profile/[uid]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const Page = () => {
```
Comment (shahdivyank) [inline src/app/director/[program]/profile/[uid]/page.tsx]: do not add the pages yet, just need the navigation

File: src/data/director/navigation.tsx [hunk 2]
```diff
@@ -1,6 +1,30 @@
 export const links = [
   {
-    name: "Interviews",
-    link: "/admin/interviews",
+    name: "Director",
+    link: "",
+  },
+  {
+    name: "Applications",
+    link: "/director/:program/application",
```
Comment (shahdivyank) [inline src/data/director/navigation.tsx]: if the link is dynamic, we should instead make each link a function which returns a string for the dynamically generated url

File: src/app/director/layout.tsx [hunk 3]
```diff
@@ -6,9 +9,16 @@ type LayoutProps = {
 };
 
 const Layout = ({ children }: LayoutProps) => {
+  const pathname = usePathname();
+  const segments = pathname.split("/");
+  const navigation = links.map(({ name, link }) => ({
+    name,
+    link: link(segments[2], segments[4]),
```
Comment (shahdivyank) [inline src/app/director/layout.tsx]: what happens when segments 2 and 4 dont exist? there should be a fallback

File: src/data/director/navigation.tsx [hunk 4]
```diff
@@ -1,6 +1,31 @@
 export const links = [
   {
-    name: "Interviews",
-    link: "/admin/interviews",
+    name: "Director",
+    link: () => "",
+  },
+  {
+    name: "Applications",
+    link: (program: string) => `/director/${program}/application`,
```
Comment (shahdivyank) [inline src/data/director/navigation.tsx]: prolly want all the functions to be the same parameter wise so it gives less problems

File: src/app/director/layout.tsx [hunk 5]
```diff
@@ -1,14 +1,29 @@
...
+      isStaticRoute ? "Ignite" : segments[2] || "Ignite",
+      segments[4] || "guest",
+    ),
+  }));
+
   return (
-    <div className="h-full">
-      <Navigation links={links} />
+    <div className="flex h-full">
+      <Navigation links={navigation} />
```
Comment (shahdivyank) [inline src/app/director/layout.tsx]: logic should be handled inside navigation

**Review by seanquiambao — APPROVED** [review]

**Pull request merged**